core/runtime/common/stack_limiter.cpp

```diff
@@ -0,0 +1,848 @@
+/**
+ * Copyright Quadrivium LLC
+ * All Rights Reserved
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "runtime/common/stack_limiter.hpp"
+
+#include "common/visitor.hpp"
+#include "log/logger.hpp"
+#include "log/profiling_logger.hpp"
+
+#include <wabt/binary-reader-ir.h>
+#include <wabt/binary-reader.h>
+#include <wabt/binary-writer.h>
+#include <wabt/ir.h>
+#include <wabt/validator.h>
+
+namespace kagome::runtime {
+
+  class Stream final : public wabt::Stream {
+   public:
+   protected:
+    wabt::Result WriteDataImpl(size_t offset,
+                               const void *data,
+                               size_t size) override {
+      if (offset + size > bytes.size()) {
+        bytes.resize(offset + size);
+      }
+      std::copy_n(reinterpret_cast<const uint8_t *>(data),
+                  size,
+                  bytes.begin() + offset);
+      return wabt::Result::Ok;
+    }
+
+    wabt::Result MoveDataImpl(size_t dst_offset,
+                              size_t src_offset,
+                              size_t size) override {
+      if (dst_offset + size < bytes.size()) {
+        bytes.resize(dst_offset + size);
+      }
+      std::copy_n(bytes.begin() + src_offset, size, bytes.begin() + dst_offset);
```

Comment:
`memmove`

Reply:
copy_n is compiled to memmove on both Clang and GCC.

Reply:
Never mind though, I discovered I can just use wabt::MemoryStream.